OBSDOCS-868: Release notes for Distributed Tracing 3.2 (#75505)
openshift/openshift-docs: observability/distr_tracing/distr_tracing_rn/distr-tracing-rn-3-2.adoc

:_mod-docs-content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="distributed-tracing-rn-3-2"]
= Release notes for {DTProductName} 3.2
:context: distributed-tracing-rn-3-2

toc::[]

include::modules/distr-tracing-product-overview.adoc[leveloffset=+1]

You can use the {DTShortName} xref:../../otel/otel-forwarding.adoc#otel-forwarding-traces[in combination with] the xref:../../otel/otel-installing.adoc#install-otel[{OTELName}].

This release of the {DTProductName} includes the {TempoName} and the deprecated {JaegerName}.

////
[id="distributed-tracing-rn_3-2_cves"]

> == CVEs

Reviewer: based on the past RN for minor versions it seems we skip `CVE` sections

=== Technology Preview features

This update introduces the following Technology Preview feature:

* Support for the Tempo monolithic deployment.

:FeatureName: The Tempo monolithic deployment
include::snippets/technology-preview.adoc[leveloffset=+1]

[id="distributed-tracing-rn_3-2_tempo-release-notes_new-features-and-enhancements"]
=== New features and enhancements

This update introduces the following enhancements:

* {TempoName} 3.2 is based on the open source link:https://grafana.com/oss/tempo/[Grafana Tempo] 2.4.1.
* Allowing the overriding of resources per component.

////
[id="distributed-tracing-rn_3-2_tempo-release-notes_deprecated-functionality"]
=== Deprecated functionality

In the {TempoName} 3.2, ???
////

////
[id="distributed-tracing-rn_3-2_tempo-release-notes_removal-notice"]
=== Removal notice

In {TempoName} 3.2, the FEATURE has been removed. Bug fixes and support are provided only through the end of the 3.? lifecycle. As an alternative to the FEATURE for USE CASE, you can use the ALTERNATIVE instead.
////

[id="distributed-tracing-rn_3-2_tempo-release-notes_bug-fixes"]
=== Bug fixes

> This update introduces the following bug fixes:

Reviewer 2: We need to add this bug fix for Tempo. https://issues.redhat.com/browse/TRACING-4087

Reviewer: This does not seem to be reported by customers. Therefore I would skip it.

* Before this update, the `query-frontend` pod might get stopped when out-of-memory (OOM) as a result of searching a large trace. With this update, resource limits can be set to prevent this issue. (link:https://issues.redhat.com/browse/TRACING-4009[TRACING-4009])

[id="distributed-tracing-rn_3-2_tempo-release-notes_known-issues"]
=== Known issues

There is currently a known issue:

* Currently, the {TempoShortName} fails on the IBM Z (`s390x`) architecture. (link:https://issues.redhat.com/browse/TRACING-3545[TRACING-3545])

[id="distributed-tracing-rn_3-2_jaeger-release-notes"]
== {JaegerName}

The {JaegerName} is provided through the {JaegerOperator} Operator.

[IMPORTANT]
====
Jaeger does not use FIPS validated cryptographic modules.
====

[id="distributed-tracing-rn_3-2_jaeger-release-notes_support-for-elasticsearch-operator"]
=== Support for {es-op}

{JaegerName} 3.2 is supported for use with the {es-op} 5.6, 5.7, and 5.8.

[id="distributed-tracing-rn_3-2_jaeger-release-notes_deprecated-functionality"]
=== Deprecated functionality

In the {DTProductName} 3.2, Jaeger and support for Elasticsearch remain deprecated, and both are planned to be removed in a future release.

> Red Hat will provide critical and above CVE bug fixes and support for these components during the current release lifecycle, but these components will no longer receive feature enhancements.

Reviewer 3: ```suggestion
Red Hat provides critical and above CVE bug fixes and support for these components during the current release lifecycle, but these components are no longer receiving feature enhancements.
```

Reviewer 3: A note to remove future tense wording. I'm also wondering about "and above" here. 
![image](https://github.com/openshift/openshift-docs/assets/77501157/66fe704f-79a7-4f83-aa87-b6338ea3e535)

Reviewer 3: IBMSG suggests other words.

The {TempoOperator} and the {OTELName} are the preferred Operators for distributed tracing collection and storage.

> The OpenTelemetry and Tempo distributed tracing stack is to be adopted by all users because this will be the stack that will be enhanced going forward.

Reviewer 3: ```suggestion
Broad adoption of the OpenTelemetry and Tempo distributed tracing stack by all users is expected because this stack that will be enhanced going forward.
```

Reviewer 3: A suggestion to avoid future tense, per https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense

In the {DTProductName} 3.2, the Jaeger agent is deprecated and planned to be removed in the following release.

> Red Hat will provide bug fixes and support for the Jaeger agent during the current release lifecycle, but the Jaeger agent will no longer receive enhancements and will be removed.

Reviewer 3: ```suggestion
Red Hat provides bug fixes and support for the Jaeger agent during the current release lifecycle, but the Jaeger agent is no longer receive enhancements and is planned for removal.
```

Reviewer 3: See, https://redhat-documentation.github.io/supplementary-style-guide/#statements-about-the-future for more information on how to mention future removal.

The OpenTelemetry Collector provided by the {OTELName} is the preferred Operator for injecting the trace collector agent.

////
[id="distributed-tracing-rn_3-2_jaeger-release-notes_removal-notice"]
=== Removal notice

> In {DTProductName} 3.2, the FEATURE has been removed. Bug fixes and support are provided only through the end of the 3.? lifecycle. As an alternative to the FEATURE for USE CASE, you can use the ALTERNATIVE instead.

Reviewer 3: I think there's some placeholder text here that needs to be removed and same question as above about 3.?

////

[id="distributed-tracing-rn_3-2_jaeger-release-notes_new-features-and-enhancements"]
=== New features and enhancements

This update introduces the following enhancements for the {JaegerShortName}:

* {JaegerName} 3.2 is based on the open source link:https://www.jaegertracing.io/[Jaeger] release 1.57.0.

[id="distributed-tracing-rn_3-2_jaeger-release-notes_known-issues"]
=== Known issues

> There is currently a known issue:

Reviewer 4: ![Screenshot 2024-05-23 at 2 15 05 PM](https://github.com/openshift/openshift-docs/assets/96192119/b4c8c2fc-acc9-426a-acfd-741bb596a165)

> * Currently, Apache Spark is not supported.

Reviewer 2: This known issue from the past releases is present and unresolved. 

Currently, the distributed tracing platform (Tempo) fails on the IBM Z (s390x) architecture. ([TRACING-3545](https://issues.redhat.com/browse/TRACING-3545))
